moveit_planners/trajopt/trajopt_interface/src/trajopt_interface.cpp

```diff
@@ -117,27 +117,39 @@ bool TrajOptInterface::solve(const planning_scene::PlanningSceneConstPtr& planni
   }
 
   ROS_INFO(" ======================================= Create ProblemInfo");
-  ProblemInfo problem_info(planning_scene, req.group_name);
-
+  trajopt::ProblemInfo problem_info(planning_scene, req.group_name);
   setProblemInfoParam(problem_info);
 
-  ROS_INFO(" ======================================= Populate init info, hard-coded");
-  // TODO: init info should be defined by user. To this end, we need to add seed trajectories to MotionPlanRequest.
-  // JOINT_INTERPOLATED: data is the current joint values
-  // GIVEN_TRAJ: data is the joint values of the current state copied to all timesteps
-  Eigen::VectorXd current_joint_values_eigen(dof);
-  for (int joint_index = 0; joint_index < dof; ++joint_index)
+  ROS_INFO(" ======================================= Populate init info");
+  // For type JOINT_INTERPOLATED, we need the configuration of the robot at one state (which should be the goal), we do
+  // not need the robot configuration along the whole trajectory. That is why we need one index which is 0 for
+  // "points[]"
+  if (problem_info.init_info.type == trajopt::InitInfo::JOINT_INTERPOLATED)
   {
-    current_joint_values_eigen(joint_index) = current_joint_values[joint_index];
-  }
+    Eigen::VectorXd initial_joint_values_eigen(dof);
+    for (int joint_index = 0; joint_index < dof; ++joint_index)
+    {
+      initial_joint_values_eigen(joint_index) =
+          req.reference_trajectories[0].joint_trajectory[0].points[0].positions[joint_index];
+    }
 
-  if (problem_info.init_info.type == InitInfo::JOINT_INTERPOLATED)
-  {
-    problem_info.init_info.data = current_joint_values_eigen;
+    problem_info.init_info.data = initial_joint_values_eigen;
   }
-  else if (problem_info.init_info.type == InitInfo::GIVEN_TRAJ)
+  else if (problem_info.init_info.type == trajopt::InitInfo::GIVEN_TRAJ)
   {
-    problem_info.init_info.data = current_joint_values_eigen.transpose().replicate(problem_info.basic_info.n_steps, 1);
+    int num_steps = problem_info.basic_info.n_steps;
+    trajopt::TrajArray init_traj;
+    init_traj.resize(num_steps, dof);
+    for (std::size_t step = 0; step < num_steps; ++step)
+    {
+      for (std::size_t joint_index = 0; joint_index < dof; ++joint_index)
+      {
+        init_traj(step, joint_index) =
+            req.reference_trajectories[0].joint_trajectory[0].points[step].positions[joint_index];
```

Comment:
I called the TrajOpt in `demo.launch` of `panda_moveit_config` and set the start and goal on the GUI.
When the plan is executed on the GUI, `reference_trajectories` is not defined.
So, trajopt planner is core dumped.
https://github.com/ros-planning/panda_moveit_config

```bash
roslaunch panda_moveit_config demo.launch
```
![Uploading Screenshot 2019-11-20 15:44:17.png…]()


There is no problem in the following launch of moveit_tutorials. Therefore, reference_trajectories is defined.

```Bash
roslaunch moveit_tutorials trajopt_example_launch.launch
```

`moveit_tutorials/doc/trajopt_planner/src/trajopt_example.cpp`

```cpp
  for (std::size_t step_index = 1; step_index <= N_STEPS; ++step_index)
  {
    std::vector<double> joint_values;
    for (int dof_index = 0; dof_index < N_DOF; ++dof_index)
    {
      double joint_value = current_joint_values[dof_index] + step_index * dt_vector[dof_index];
      joint_values.push_back(joint_value);
    }
    req.reference_trajectories[0].joint_trajectory[0].joint_names = joint_names;
    req.reference_trajectories[0].joint_trajectory[0].points[step_index].positions = joint_values;
  }
```

Should the above implementation be included in `trajopt_interface.cpp`?

Reply:
Right, in motion planning display in rviz, we do not set any reference traj. When using the gui, If the `init_info.type` in yaml file is set to `STATIONARY`, you would no get the error as this type does not need any reference traj. Now, I think there are two approaches to handle this. 
1- add an option in the gui that gets the reference traj from the user. 
2- warn the user that they need to set `init_info.type` in the yaml file to `STATIONARY` (when they are using the gui).

I work on the second  approach for now.

Reply:
Thank you for your reply. 
I did not understand mean `STATIONARY` in the yaml file. sorry.
I agree the second approach.